app/code/Magento/Customer/Model/AccountManagement.php

```diff
@@ -855,9 +864,33 @@ public function createAccount(CustomerInterface $customer, $password = null, $re
         } else {
             $hash = null;
         }
+        //If Customer group id specified in the request, Check the group id exist in Magento.
+        if ($customer->getGroupId()) {
+            if (!$this->checkCustomerGroupExist($customer->getGroupId())) {
+                throw new LocalizedException(
+                    __("The group id specified is invalid. Please specify valid group id.")
```

Comment:
```suggestion
                    __("The specified group id does not exist. Please specify a valid group id.")
```

Reply:
@rogyar Committed the changes. Please review it.